x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/pages/form.tsx

```diff
@@ -156,3 +160,84 @@ export const useRuleIndexPattern = ({
   }, [dataSourceType, isIndexPatternLoading, data, dataViewId, initIndexPattern]);
   return { indexPattern, isIndexPatternLoading, browserFields };
 };
+
+export interface UseRuleFormsErrors {
+  defineStepForm?: FormHook<DefineStepRule, DefineStepRule>;
+  aboutStepForm?: FormHook<AboutStepRule, AboutStepRule>;
+  scheduleStepForm?: FormHook<ScheduleStepRule, ScheduleStepRule>;
+  actionsStepForm?: FormHook<ActionsStepRule, ActionsStepRule>;
+}
+
+const getFieldErrorMessages = (fieldError: ValidationError) => {
+  if (fieldError.message.length > 0) {
+    return [fieldError.message];
+  } else if (Array.isArray(fieldError.messages)) {
+    return fieldError.messages.map((message) => JSON.stringify(message));
+  }
+  return [];
+};
+
+const NON_BLOCKING_QUERY_BAR_ERROR_CODES = [
+  ESQL_ERROR_CODES.INVALID_ESQL,
+  EQL_ERROR_CODES.FAILED_REQUEST,
+  EQL_ERROR_CODES.INVALID_EQL,
+  EQL_ERROR_CODES.MISSING_DATA_SOURCE,
+];
+
+const isNonBlockingQueryBarErrorCode = (errorCode?: string) => {
+  return !!NON_BLOCKING_QUERY_BAR_ERROR_CODES.find((code) => code === errorCode);
+};
+
+const NON_BLOCKING_ERROR_CODES = [...NON_BLOCKING_QUERY_BAR_ERROR_CODES];
+
+const isNonBlockingErrorCode = (errorCode?: string) => {
+  return !!NON_BLOCKING_ERROR_CODES.find((code) => code === errorCode);
+};
+
+const transformValidationError = ({
+  errorCode,
+  errorMessage,
+}: {
+  errorCode?: string;
+  errorMessage: string;
+}) => {
+  if (isNonBlockingQueryBarErrorCode(errorCode)) {
+    return i18n.QUERY_BAR_VALIDATION_ERROR(errorMessage);
+  }
+  return errorMessage;
+};
+
+export const useRuleFormsErrors = () => {
+  const getRuleFormsErrors = useCallback(
```

Comment:
would be great to have unit tests for this function/hook if possible

Reply:
added some basic tests